src/jobs/keys-update/keys-update.service.ts

```diff
@@ -90,119 +86,144 @@ export class KeysUpdateService {
-
-      await Promise.all(
-        this.stakingModules.map(async (stakingModule) => {
-          if (stakingModule.type === STAKING_MODULE_TYPE.CURATED_ONCHAIN_V1_TYPE) {
-            this.logger.log('Start updating curated keys');
-            await this.curatedModuleService.updateKeys(blockHash);
-          }
-        }),
-      );
-
-      // Update cached data to quick access
-      await this.updateMetricsCache();
+      const meta = await this.update();
+      await this.updateMetrics();
 
-      this.setMetrics();
+      if (meta) {
+        this.lastBlockNumber = meta.number;
+        this.lastTimestampSec = meta.timestamp;
+      }
 
-      // update finished
       // clear timeout
       this.checkKeysUpdateTimeout();
     });
   }
 
   /**
-   * Update cache data above
+   * Update keys of staking modules
+   * @returns Number, hash and timestamp of execution layer block
    */
-  @Trace(METRICS_TIMEOUT)
-  private async updateMetricsCache() {
-    const meta = await this.updateCuratedMetricsCache();
-    // update meta
-    // timestamp and block number is common for all modules
-    this.lastTimestampSec = meta?.timestamp ?? this.lastTimestampSec;
-    this.lastBlockNumber = meta?.blockNumber ?? this.lastBlockNumber;
-  }
+  public async update(): Promise<{ number: number; hash: string; timestamp: number } | undefined> {
+    // reading latest block from blockchain
+    const currElMeta = await this.executionProvider.getBlock('latest');
+    // read from database last execution layer data
+    const prevElMeta = await this.elMetaStorage.get();
+
+    if (prevElMeta && prevElMeta?.blockNumber > currElMeta.number) {
+      this.logger.warn('Previous data is newer than current data', prevElMeta);
+      console.log('curr', currElMeta);
+      return;
+    }
 
-  private async updateCuratedMetricsCache(): Promise<RegistryMeta | null> {
-    const { operators, meta } = await this.curatedModuleService.getOperatorsWithMeta();
-    // should set for each module
-    this.curatedNonce = meta?.keysOpIndex ?? this.curatedNonce;
-    // update curated module operators map
-    this.curatedOperators = operators;
+    // TODO: еcли была реорганизация, может ли currElMeta.number быть меньше и нам надо обновиться ?
 
-    return meta;
-  }
+    const storageModules = await this.srModulesStorage.findAll();
+    // get staking router modules from SR contract
+    const modules = await this.stakingRouterFetchService.getStakingModules({ blockHash: currElMeta.hash });
 
-  /**
-   * Updates prometheus metrics
-   */
-  private setMetrics() {
-    // common metrics
-    if (this.lastTimestampSec) {
-      this.prometheusService.registryLastUpdate.set(this.lastTimestampSec);
-    }
-    if (this.lastBlockNumber) {
-      this.prometheusService.registryBlockNumber.set(this.lastBlockNumber);
+    // TODO: is it correct that i use here modules from blockchain instead of storage
+
+    if (this.modulesWereDeleted(modules, storageModules)) {
```

Comment:
That's strange check.
If modules can be deleted in blockchain this is wrong.
If can not than this check is redundant